front/app/components/SignUpIn/SignUp/Consent.tsx

```diff
@@ -77,71 +79,123 @@ const Consent = memo(
       onPrivacyAcceptedChange(!privacyPolicyAccepted);
     };
 
-    return (
-      <Container className={className}>
-        <CheckboxWrapper id="e2e-terms-and-conditions-container">
-          <Checkbox
-            className="e2e-terms-and-conditions"
-            size="20px"
-            checked={termsAndConditionsAccepted}
-            onChange={handleTermsAndConditionsOnChange}
-            label={
-              <ConsentText>
-                <FormattedMessage
-                  {...messages.iHaveReadAndAgreeTo}
-                  values={{
-                    link: (
-                      <Link target="_blank" to="/pages/terms-and-conditions">
-                        <FormattedMessage {...messages.theTermsAndConditions} />
-                      </Link>
-                    ),
-                  }}
-                />
-              </ConsentText>
-            }
-          />
-          <Error
-            text={
-              termsAndConditionsError ? formatMessage(messages.tacError) : null
-            }
-          />
-        </CheckboxWrapper>
-
-        <CheckboxWrapper id="e2e-privacy-container">
-          <Checkbox
-            className="e2e-privacy-checkbox"
-            size="20px"
-            checked={privacyPolicyAccepted}
-            onChange={handlePrivacyPolicyOnChange}
-            label={
-              <ConsentText>
-                <FormattedMessage
-                  {...messages.iHaveReadAndAgreeTo}
-                  values={{
-                    link: (
-                      <Link target="_blank" to="/pages/privacy-policy">
-                        <FormattedMessage {...messages.thePrivacyPolicy} />
-                      </Link>
-                    ),
-                  }}
-                />
-              </ConsentText>
-            }
-          />
-          <Error
-            text={
-              privacyPolicyError
-                ? formatMessage(messages.privacyPolicyNotAcceptedError)
-                : null
-            }
-          />
-        </CheckboxWrapper>
-
-        <ConsentText>
-          <FormattedMessage {...messages.emailConsent} />
-        </ConsentText>
-      </Container>
-    );
+    if (authProvider === 'id_vienna_saml') {
```

Comment:
Maybe everything that is rendered if `authProvider === 'id_vienna_saml` should be an `Outlet` living inside of the `id_vienna_saml` module? If that's really complicated it's ok, I know how messed up the sign up flow + module set up is, but would be nice

Reply:
I am for sure not able to do this on my own. Vienna needs the button to go live on Monday morning. If you request this, can we go ahead with the current implementation and refactor later? I'm happy to do it, but we are running out of a time with this a litle.

Also, I'd need help because I don't know if we prefer code duplication or somehow extract the shared behaviour.